app/logic/fileHandler.py

```diff
@@ -12,41 +12,55 @@ def __init__(self,files=None, courseId=None, eventId=None):
         if courseId:
             self.path = os.path.join(self.path, app.config['files']['course_attachment_path'], str(courseId))
         elif eventId:
-            self.path = os.path.join(self.path, app.config['files']['event_attachment_path'], str(eventId))
+            self.path = os.path.join(self.path, app.config['files']['event_attachment_path'])       #Removed event ID from how the file path is being saved.
+        
         try:
-            os.makedirs(self.path)
+            extraDir = str(self.eventId) if self.eventId else ""
+            os.makedirs(os.path.join(self.path, extraDir))                                          #Changed the path where the directory is created.
         except:
             print("Directory exists.")
 
-    def getFileFullPath(self, newfile = None):
+    def getFileFullPath(self, newfilename = ''):
         """
         This creates the directory/path for the object from the "Choose File" input in the create event and edit event.
         :returns: directory path for attachment
         """
+
+        # Added the eventID of the first recurring event to track the file path for subsequent recurring events.
         try:
             # tries to create the full path of the files location and passes if
             # the directories already exist or there is no attachment
-            filePath=(os.path.join(self.path, newfile.filename))
+            filePath=(os.path.join(self.path, "".join(newfilename)))  # This line adds the eventID of the first recurring event.
```

Comment:
The comment doesn't make much sense here. This function doesn't know anything about event IDs.